src/tree.js

```diff
@@ -266,6 +266,23 @@ export let flattenTree =
 
 export let flatLeaves = (next = traverse) => _.reject(next)
 
+export const find =
+  (next = traverse) =>
+  (it, tree) => {
+    let result = {}
+    try {
+      walk(next)((node, key, parents, parentsKeys) => {
+        if (_.iteratee(it)(node)) {
+          result = { node, key, parents, parentsKeys }
+          throw new Error('FOUND')
```

Comment:
You don’t need an error - returning truthy from walk will halt iteration

Reply:
This will break quicker without backtracking though